sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/Transforms.java

```diff
@@ -789,4 +792,25 @@ static PathDeletedItem toPathDeletedItem(BlobItemInternal blobItem) {
     static PathDeletedItem toPathDeletedItem(BlobPrefix blobPrefix) {
         return new PathDeletedItem(blobPrefix.getName(), true, null, null, null);
     }
+
+    private static EncryptionAlgorithmType toBlobEncryptionAlgorithmType(
+        com.azure.storage.file.datalake.models.EncryptionAlgorithmType type) {
+        return EncryptionAlgorithmType.valueOf(type.toString());
+    }
+
+    static CustomerProvidedKey toBlobCustomerProvidedKey(
```

Comment:
Looks like it's split about 50/50 in terms of number of methods, which seems reasonable because if it's getting converted one way on the way out, it's probably undoing that on the way back. Might depend on the workload which direction is used more.

That's a good thought about performance, though. Given the split, it may be good to just pick a direction and say all datalake objects should just be wrappers of the blob objects

Reply:
I wonder if for features like this we should create types in common and have blob types wrap common type.

Reply:
How do you define "features like this"?

Reply:
model/options types that exists in more than one sdk. Like `ParallelTransferOptions` options. If cpk has potential to expand beyond blobs and datalake then this might be a good place for related types. On the other hand I guess it's not very likely to happen, so not polluting queues and files for now is better strategy. Please disregard that comment.